src/character.cpp

```diff
@@ -2124,13 +2124,19 @@ hp_part Character::body_window( const std::string &menu_header,
             std::string h_bar = get_hp_bar( hp, maximal_hp, false ).first;
             nc_color h_bar_col = get_hp_bar( hp, maximal_hp, false ).second;
 
-            return tag_colored_string( h_bar, h_bar_col ) + tag_colored_string( std::string( 5 - h_bar.size(), '.' ), c_white );
+            return tag_colored_string( h_bar, h_bar_col ) + tag_colored_string( std::string( 5 - h_bar.size(),
+                    '.' ), c_white );
         }
     };
 
     uilist bmenu;
     bmenu.desc_enabled = true;
-    bmenu.text = string_format( _( "Select a body part for: %s" ), menu_header );
+    if( menu_header != disp_name() ) {
+        bmenu.text = string_format( _( "Select a body part for: %s" ), menu_header );
```

Comment:
And since `Character::body_window` now contains quite a few strings related to healing effects, I guess the better way is to move wound inspection menu to a separate function.

Reply:
Since there are only few places to fix, I followed your idea and pushed header construction as early as possible.